Firestore/core/src/core/firestore_client.cc

```diff
@@ -107,13 +109,14 @@ static const size_t kMaxConcurrentLimboResolutions = 100;
 std::shared_ptr<FirestoreClient> FirestoreClient::Create(
     const DatabaseInfo& database_info,
     const api::Settings& settings,
-    std::shared_ptr<auth::CredentialsProvider> credentials_provider,
-    std::shared_ptr<util::Executor> user_executor,
-    std::shared_ptr<util::AsyncQueue> worker_queue) {
+    std::shared_ptr<CredentialsProvider> credentials_provider,
+    std::shared_ptr<Executor> user_executor,
+    std::shared_ptr<AsyncQueue> worker_queue,
+    std::unique_ptr<FirebaseMetadataProvider> firebase_metadata_provider) {
   // Have to use `new` because `make_shared` cannot access private constructor.
-  std::shared_ptr<FirestoreClient> shared_client(
-      new FirestoreClient(database_info, std::move(credentials_provider),
-                          std::move(user_executor), std::move(worker_queue)));
+  std::shared_ptr<FirestoreClient> shared_client(new FirestoreClient(
```

Comment:
curious: any reason not to use `make_shared()` here?

Reply:
`make_shared` requires a public constructor but we're forcing callers to use `FirestoreClient::Create` and making the constructor private.

Reply:
makes sense, thanks for explaining